armor-write/src/main/java/com/rapid7/armor/store/S3WriteStore.java

```diff
@@ -725,6 +720,29 @@ private void trackTenant(String tenant) {
     }
   }
 
+  private List<S3ObjectSummary> getShardObjects(ShardId shardId) {
+    List<S3ObjectSummary> objectsToCopy = new ArrayList<>();
+    
+    for (int i = 0; i < 10; i++) {
+      DistXact currentValues = getCurrentValues(shardId);
+      if (currentValues != null) {
+        String currentShardKey = PathBuilder.buildPath(shardId.shardIdPath(), currentValues.getCurrent());
+        ListObjectsV2Request request = new ListObjectsV2Request().withBucketName(bucket).withPrefix(currentShardKey);
+        ListObjectsV2Result result;
+        do {
+          result = s3Client.listObjectsV2(request);
+          objectsToCopy.addAll(result.getObjectSummaries());
+          request.setContinuationToken(result.getNextContinuationToken());
+        } while (result.isTruncated());
+      }
+      
+      if (!objectsToCopy.isEmpty()) {
```

Comment:
I don't think its empty. I think they only thing you need to do is put this in a try-catch for this everything in this for loop. we can talk about this.